### PR TITLE
Add example of proxying

### PR DESCRIPTION
**Pull request by aulorbe**
## Problem

The community has requested clearer direction regarding using proxy servers and other http-related customizations with our Typescript client. 

Asana ticket https://app.asana.com/0/1203260648987893/1205491455268495/f

## Solution

Add proxying section to README + clarify some code comments.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [x] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan

Describe specific steps for validating this change.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205491455268495

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,6 @@ ts-compilation-test/lib/
 .vscode
 .DS_Store
 scratch
+
```
Comment (aulorbe) [inline .gitignore:18]: Just housekeeping

File: README.md [hunk 2]
```diff
@@ -77,6 +77,65 @@ const pc = new Pinecone({
...
+const cert = fs.readFileSync('path-to-your-mitm-proxy-cert-pem-file');
+
+const client = new ProxyAgent({
+  uri: '<your proxy server URI>',
+  requestTls: {
+    port: '<your proxy server port>',
+    ca: cert,
+    host: '<your proxy server host>',
+  },
+});
```
Comment (austin-denoble) [inline README.md:109]: Nice job with the example documentation. This feels like a succinct and reasonable way to configure a proxy. 👍

**Review by austin-denoble — APPROVED** [review]
LGTM, nice work sorting out the proper way to configure this through the client. We should sync on getting setup with pclocal because I think it'll look similar to this.

File: README.md (hunk 2 above)
Comment (aulorbe) [reply README.md:109]: thanks!

**Pull request merged**